packages/merge-styles/src/mergeStyleSets.test.ts

```diff
@@ -196,14 +195,118 @@ describe('mergeStyleSets', () => {
   });
 
   it('can merge two sets with class names', () => {
-    const styleSet1: ITestClasses = mergeStyleSets({
+    const styleSet1 = mergeStyleSets({
       root: ['ms-Foo', { background: 'red' }]
     });
-    const styleSet2: ITestClasses = mergeStyleSets(styleSet1, {
+    const styleSet2 = mergeStyleSets(styleSet1, {
       root: ['ms-Bar', { background: 'green' }]
     });
 
     expect(styleSet2).toEqual({ root: 'ms-Foo ms-Bar root-1', subComponentStyles: {} });
     expect(_stylesheet.getRules()).toEqual('.root-0{background:red;}' + '.root-1{background:green;}');
   });
+
+  describe('typings tests', () => {
+    interface ISubComponentStyles extends IStyleSet<ISubComponentStyles> {
+      root: IStyle;
+    }
+
+    interface ISubComponentStyleProps {
+      isCollapsed: boolean;
+    }
+
+    interface IStyles extends IStyleSet<IStyles> {
+      root: IStyle;
+      subComponentStyles: {
+        button: IStyleFunctionOrObject<ISubComponentStyleProps, IStyleSet<ISubComponentStyles>>;
+      };
+    }
+
+    interface IStylesWithStyleObjectAsSubCommponent extends IStyleSet<IStyles> {
+      root: IStyle;
+      subComponentStyles: {
+        button: Partial<IStyleSet<ISubComponentStyles>>;
+      };
+    }
+
+    /** Button component as of test writing has this interface. */
+    const LegacySubComponent: (props: { styles: Partial<ISubComponentStyles> }) => any = (props: {
+      styles: Partial<ISubComponentStyles>;
+    }) => {
+      return;
+    };
+
+    const SubComponent: (
+      props: { styles: IStyleFunctionOrObject<ISubComponentStyleProps, ISubComponentStyles> }
+    ) => any = (props: { styles: IStyleFunctionOrObject<ISubComponentStyleProps, ISubComponentStyles> }) => {
+      return;
+    };
+
+    const getStyles = (): IStyles => ({
+      root: {
+        background: 'red'
+      },
+      subComponentStyles: {
+        button: () => ({
+          root: {
+            background: 'green'
+          }
+        })
+      }
+    });
+
+    it('IStyleSet/IProcessedStyleSet should work with standard sub components', () => {
+      const classNames = mergeStyleSets<IStyles>(getStyles());
+
+      SubComponent({ styles: classNames.subComponentStyles.button });
+
+      // this test primarily tests that the lines above do not result in a Typescript error.
+      expect.assertions(0);
```

Comment:
Could `it` just be renamed so it's not an executable test case while still having typings generate an error?

Reply:
This is to exercise compiler though...

Reply:
Renamed how? I actually would like it to show up as an executable test actually that reflects in the jest log, and also unit test count, since these are actually testing things :)

Reply:
@kenotron These tests would fail the build without Jest actually running them, but I see Cliff's point about stat tracking